pkg/scheduler/schedule_one.go

```diff
@@ -306,6 +306,13 @@ func (sched *Scheduler) bindingCycle(
 		return status
 	}
 
+	// Any failures after this point cannot lead to the Pod being considered unschedulable.
+	// We define the Pod as "unschedulable" only when Pods are rejected at specific extension points, and PreBind is the last one in the scheduling/binding cycle.
+	//
+	// We can call Done() here because
+	// we can free the cluster events stored in the scheduling queue sonner, which is worth for busy clusters memory consumption wise.
+	sched.SchedulingQueue.Done(assumedPod.UID)
```

Comment:
Shouldn't `Done` call be removed from [there](https://github.com/kubernetes/kubernetes/blob/01eb9f47541156a38e4b28759977c06f07f08012/pkg/scheduler/schedule_one.go#L132)?

Reply:
Correct, that's a reductant Done.
Removed.